ambry-api/src/main/java/com/github/ambry/quota/QuotaSource.java

```diff
@@ -63,8 +65,8 @@
 
   /**
    * Get the percent usage of system resources relevant to the {@link QuotaName} specified. The exact definition of
-   * system resource is left for the source and enforcer to determine based on the type of quota. Return {@literal -1}
-   * if system resource usage doesn't make sense for a combination of quota, source or enforcer.
+   * system resource is left for the implementation to determine.
+   * Return {@literal -1} if system resource usage doesn't make sense.
```

Comment:
What do you mean by "if system resource usage doesn't make sense" ? What system resources do not make sense and why?

Same comment repeats in mant places

Reply:
I have updated this comment with "if the implementation doesn't need to track any system resource".

Note that tracking system resource along with quota is very implementation specific, so its hard to be more specific here. For example, in Ambry, the current storage quota implementation doesn't track the overall storage usage in a cluster (unlike bandwidth quota which does track overall bandwidth usage in a node). As a side-effect, a call to `isQuotaExceedAllowed` for storage quota's `QuotaEnforcer` implementation, will always return false.
Another implementation of storage quota could have, maybe, taken overall storage usage in a cluster into account, to decide the return value for `isQuotaExceedAllowed`.
So in cases where implementation doesn't need to track system resources (for whatever reason), the quota source will return the specified values as mentioned in the comment.